project/SL510457/README.md

```diff
@@ -0,0 +1,82 @@
+# TourTracer : Christofides-TSP-Solver
+This proposal outlines the workflow and plans for the NYCU NSD term project in Autumn 2023.
+
+## Basic Information
+### Traveling Salesman Problem (TSP)
+TSP is a NP hard problem. Given a set of cities and the distance between every pair of cities, the problem is to find the shortest possible route that visits each city exactly once and returns back to the original city. 
+
+### Christofides algorithm
+The Christofides algorithm is an approximation algorithm for the Traveling Salesman Problem (TSP) that offers a good trade-off between solution quality and computational efficiency. It guarantees a solution within 1.5 times the optimal solution by using techniques like MST, minimum-weight perfect matching, Eulerian circuit.
+
+
+GitHub Link : [ https://github.com/SL510457/TourTracer-Christofides-TSP-Visualizer ]
+
+
+
+## Problem to Solve
+Given a graph with nodes and weighted edges, find a Hamiltonian cycle in the graph that minimizes the total distance traveled. 
```

Comment:
How do you measure the distance?  How large your graph is?  How do you plan to create or obtain the test data?

Reply:
I think I would have the user input the coordinate position (possibly in 2D or 3D), and then use the calculated distance between these coordinates as a measure of their weight.